2-mandatory/2-water-bottle.js

```diff
@@ -17,20 +17,43 @@ You have to implement the missing features according to the specification.
 // Here is your starting point:
 let bottle = {
   volume: 0,
+
   fillUp: function () {
-    // calling this function should completely fill your bottle (volume = 100);
+     // calling this function should completely fill your bottle (volume = 100);
+     this.volume = 100;
   },
+
   pour: function () {
-    // calling this function should increase your bottle volume by 10 units;
+     // calling this function should increase your bottle volume by 10 units;
+     if (this.volume <= 90) {
+        this.volume += 10;
+     } else if (this.volume > 90 && this.volume < 100) {
+        this.volume = 100;
+     } else {
+        this.volume = this.volume;
+     }
   },
+
   drink: function () {
-    // calling this function should decrease your bottle volume by 10 units;
+     // calling this function should decrease your bottle volume by 10 units;
+     this.volume = this.volume;
+     if (this.volume > 10) {
```

Comment:
This could be much simpler:

if (this.volume < 10) {
    this.volume = 10;
} else {
    this.volume = this.volume - 10;
}

Or even:

this.volume = this.volume < 10 ? 0 : this.volume -10;